docs/machine-learning/how-to-guides/inspect-intermediate-data-ml-net.md

```diff
@@ -1,75 +1,180 @@
+
+![Data Debugger Preview Object](./media/inspect-intermediate-data-ml-net/data-debugger-preview-01.png)
+
+### Preview contents of an IDataView
+
+The snippet below takes an [`IDataView`](xref:Microsoft.ML.IDataView) inspects the contents of the rows and columns.
+
 ```csharp
-// Create a new context for ML.NET operations. It can be used for exception tracking and logging,
-// as a catalog of available operations and as the source of randomness.
-var mlContext = new MLContext();
-
-// Read the data into a data view.
-var data = mlContext.Data.ReadFromTextFile<InspectedRow>(dataPath, hasHeader: true);
-
-// Start creating our processing pipeline. For now, let's just concatenate all the text columns
-// together into one.
-var pipeline = mlContext.Transforms.Concatenate("AllFeatures", "WorkClass", "Education", "MaritalStatus");
-
-// Fit our data pipeline and transform data with it.
-var transformedData = pipeline.Fit(data).Transform(data);
-
-// Extract the 'AllFeatures' column.
-// This will give the entire dataset: make sure to only take several row
-// in case the dataset is huge. The is similar to the static API, except
-// you have to specify the column name and type.
-var featureColumns =
-    transformedData
-        .GetColumn<string[]>(mlContext, "AllFeatures")
-        .Take(20)
-        .ToArray();
+// Preview IDataView
+DataDebuggerPreview dataPreview = data.Preview(maxRows:5);
 ```
+
+![Data Debugger Preview Row View](./media/inspect-intermediate-data-ml-net/data-debugger-preview-02.png)
+
+### Preview results after applying data pre-processing transformations
+
+Call the [`Preview`](xref:Microsoft.ML.DebuggerExtensions.Preview*) method on a data pre-processing [`IEstimator`](xref:Microsoft.ML.IEstimator`1) and pass in an [`IDataView`](xref:Microsoft.ML.IDataView) as a parameter to preview what the result of applying pre-processing transformations would look like on the data.
+
+```csharp
+// Define data pre-processing estimator
+IEstimator<ITransformer> dataPrepEstimator = 
+    mlContext.Transforms.Concatenate("Features", new string[] { "Size", "HistoricalPrices" })
+        .Append(mlContext.Transforms.NormalizeMinMax("Features"));
+
+// Preview results of applying pre-processing transformations
+DataDebuggerPreview dataPrepPreview = dataPrepEstimator.Preview(data,maxRows:5);
+```
+
+### Preview results after training a machine learning model
+
+Upon inspection of the first element in the `ColumnView` of `regressionModelPreview`, the contents in addition to the data pre-processing transformations also have the predicted output of the machine learning model which is in the `Score` column. 
+ 
+```csharp
+// Preview results of training sdca regression model
+IDataView transformedData = dataPrepEstimator.Fit(data).Transform(data);
+
+IEstimator<ITransformer> sdcaEstimator = mlContext.Regression.Trainers.Sdca();
+
+DataDebuggerPreview regressionModelPreview = sdcaEstimator.Preview(transformedData,maxRows:5);
```

Comment:
>Preview [](start = 59, length = 7)

Instead of Preview() you can convert this to IEnumerable and look at it. Feel free to look at our API reference samples here:

[https://github.com/dotnet/machinelearning/tree/master/docs/samples/Microsoft.ML.Samples/Dynamic](https://github.com/dotnet/machinelearning/tree/master/docs/samples/Microsoft.ML.Samples/Dynamic)

Reply:
#resolved